README.md

```diff
@@ -27,3 +27,5 @@ about it at the start of each recursive call.
 Describe your reasoning and the conclusion you've come to. Your reasoning is the
 most important part. You do not need to prove that the invariant is correct. Add
 your answer to this markdown file.
+
+- A good invariant could be that fib(0) = [0] and fib(1) = [0,1]. This works because in any case of n this is true: for n = 0 we will always return the array [0] and for n = 1 we return the array [0,1], and for any choice of n greater than these two values we will recursively fall back on these two facts to further compute fibonacci values of greater n's. 
```

Comment:
This invariant doesn't help me to prove that the implementation is correct. It only says something about the first two numbers.

Reply:
Actually sorry would a better one be that fib(n) = fib(n-1) + fib(n-2) for n > 2 since no matter which n you can choose recursively its Fibonacci value will always be the sum of the two previous ones?

Reply:
That sounds better, but what about a specific iteration `i`?

Reply:
Would it be for all iterations i > 1, fib(i) = fib(i-1) + fib(i-2) where fib(0) and fib(1) are known? Because then you would use fib(0) and fib(1) as base cases for mathematical induction of proving this holds true for any iteration of the function.

Reply:
Also apologies for the late response

Reply:
What about the array where the values are stored? Can you say something about that?

Reply:
The values would be stored in an array before we call the function for the first time

Reply:
Ok, and your invariant must say something about this array as that's what's being returned.

Reply:
Gotcha, so the invariant would be that for all iterations i > 1, fib(i) = fib(i-1) + fib(i-2), which will be returned as an array, where fib(0) and fib(1) are known and stored in their respective arrays

Reply:
There's only one array here. Talk about what's stored in there.

Reply:
Though I don’t know how to tweak that so we don’t have duplicates of fib(0) and fib(1) in the array

Reply:
Not sure what you mean. You can simply run your code and see what's stored in the array in each iteration.

Reply:
Hey sorry I thought I replied to this, but what returns is an array of fib values, where the iteration determines how many fib values are stored ie) if i = 0, we return [fib(0)], and if i = 1, [fib(0), fib(1)], etc.

Reply:
That sounds like it's going in the right direction.

Reply:
So for all iterations i >= 0, fib(i) will return an array of fib values, the size of which is i + 1

Reply:
Hey sorry for yet another late response, but what else do I need to do for this exercise?

Reply:
Your invariant doesn't say anything general at the moment, just for specific iterations.

Reply:
So generally speaking, do I talk about what's in the array before the first iteration of the recursive process?

Reply:
Could I say for an invariant that for any n, fib(n) will always be the last element of the array?

Reply:
You need to say something about the current iteration, not the first one. And fib(n) will be in there *after* the loop has run, but what about before?

Reply:
So can I say fib(i) will always be the last element of the array?

Reply:
Is it `i`, or the previous value?

Reply:
It'd be the previous value since we use recursion to determine it, meaning that will be the one at the end of the array right?

Reply:
Sounds like it's going in the right direction.